packages/SwingSet/src/liveslots/virtualReferences.js

```diff
@@ -62,48 +62,69 @@ export function makeVirtualReferenceManager(
     const exportStatus = getExportStatus(vobjID);
     if (exportStatus !== 'reachable' && refCount === 0) {
       let doMoreGC = deleteStoredRepresentation(vobjID);
-      syscall.vatstoreDelete(`vom.rc.${vobjID}`);
-      syscall.vatstoreDelete(`vom.es.${vobjID}`);
-      doMoreGC = doMoreGC || ceaseRecognition(vobjID);
+      const { slot } = parseVatSlot(vobjID);
+      syscall.vatstoreDelete(`vom.rc.${slot}`);
+      syscall.vatstoreDelete(`vom.es.${slot}`);
+      const more = ceaseRecognition(slot);
+      doMoreGC = doMoreGC || more;
       return [doMoreGC, exportStatus !== 'none'];
     }
     return [false, false];
   }
 
   function getRefCount(vobjID) {
-    const raw = syscall.vatstoreGet(`vom.rc.${vobjID}`);
+    const { slot } = parseVatSlot(vobjID);
+    const raw = syscall.vatstoreGet(`vom.rc.${slot}`);
     if (raw) {
       return Number(raw);
     } else {
       return 0;
     }
   }
 
-  function getExportStatus(vobjID) {
-    const raw = syscall.vatstoreGet(`vom.es.${vobjID}`);
-    switch (raw) {
-      case '0':
-        return 'recognizable';
-      case '1':
+  function getExportStatus(slot) {
+    const es = syscall.vatstoreGet(`vom.es.${slot}`);
+    if (es) {
+      if (es.indexOf('r') >= 0) {
```

Comment:
This "max(facetFlags)" approach feels right. I need to think about what actually happens as the parts of a cohort are getting cleaned out, though, if there's anything funny that might cause problems.